### ISSUE TITLE
Failed to save: Cannot read properties of undefined (reading 'logger')

### ISSUE THREAD

**Issue by JeppeKlitgaard** [microsoft/vscode#147929]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.66.2
- OS Version: Windows 10

Full system info:
	  Version: 1.66.2 (user setup)
	  Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
	  Date: 2022-04-11T07:46:01.075Z
	  Electron: 17.2.0
	  Chromium: 98.0.4758.109
	  Node.js: 16.13.0
	  V8: 9.8.177.11-electron.0
	  OS: Windows_NT x64 10.0.19044

I am getting this error intermittently when trying to save a Jupyter Notebook. It seems related to the underlying file being updated (by a linter, git, etc) while a cell is executing.

Developer console reveals:
![164746069-5fe9ae9c-afd0-411c-a7cb-a009ef5ab25b](https://user-images.githubusercontent.com/1814611/164747470-80b53ce1-e341-4c2a-99cd-3338b8639a91.png)

I am also getting a lot of these, but it might be unrelated:
![164746361-a032c8db-c9e4-4c14-8ac0-aaee806a5894](https://user-images.githubusercontent.com/1814611/164747450-81683b11-cd9c-4aa7-b9b2-7f0ec61356e3.png)

The full log can be found here:
[vscode-app-1650641355098.log](https://github.com/microsoft/vscode/files/8542215/vscode-app-1650641355098.log)

See also: #144614
@bpasero

**Comment by NickCrews** [comment]
This is also happening to me. It happens when I run `black` when a cell is running, which is consistent with the proposed cause that @JeppeKlitgaard suggests.

Version: 1.66.2
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:49:20.994Z (1 wk ago)
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin x64 21.4.0

I can post other logs, but not sure what would be useful for people.

**Comment by bpasero** [comment]
@JeppeKlitgaard @NickCrews a detailed step by step instruction for how to reproduce from scratch please.

**Comment by NickCrews** [comment]
Thanks @bpasero for the quick fix! if I use vscode-insiders will I get this bugfix immediately?

Just for longevity, to reproduce:

1. Install `python -m pip install black[jupyter]`
2. Run this code in a cell in `test.ipynb`:
```python
import time

# black will try to reformat by removing spaces
time.sleep( 10 )
```
3. As that is running, in a terminal run `black test.ipynb`

**Comment by bpasero** [comment]
Yes, insiders from Monday next week.

**Comment by JeppeKlitgaard** [comment]
/verified

On initial testing, it does appear to be fixed. I never found a way of reproducing it consistently, however.

**Comment by NickCrews** [comment]
My repro steps were consistent for me FYI. I just tested and it does appear fixed. Thanks!

**Comment by karray** [comment]
I had this problem saving a notebook (that had root rights) on a remote server and I didn't have permission

**Comment by NickCrews** [comment]
@karray maybe open a new issue? The problem that I had is now fixed in VScode 1.67 (April 2022), so I bet they are different causes.

**Comment by Serial-Comma** [comment]
![image](https://user-images.githubusercontent.com/47783328/167297396-81b0d8c1-9b35-4b55-88a0-64336e6a81bb.png)
I have this issue too, VScode 1.66.2 on Arch
logs: https://gist.github.com/Serial-Comma/ba0eb2ade6f13eedc27630cad65de7dc

**Comment by NickCrews** [comment]
@Serial-Comma it should be fixed in 1.67, update to that and see if you still have the problem.

**Comment by Serial-Comma** [comment]
@NickCrews Yep just updated and it saves just fine now
